engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -0,0 +1,97 @@
+package cromwell.engine.workflow.lifecycle.execution
+
+import akka.actor.{ActorRef, LoggingFSM, Props}
+import cromwell.backend.BackendJobExecutionActor._
+import cromwell.backend.{BackendInitializationData, BackendJobDescriptor, BackendJobDescriptorKey, BackendLifecycleActorFactory}
+import cromwell.core.logging.WorkflowLogging
+import cromwell.engine.workflow.lifecycle.execution.EngineJobExecutionActor._
+import cromwell.engine.workflow.lifecycle.execution.JobPreparationActor.{BackendJobPreparationFailed, BackendJobPreparationSucceeded}
+import cromwell.jobstore._
+
+object EngineJobExecutionActor {
+  /** States */
+  sealed trait EngineJobExecutionActorState
+  case object Pending extends EngineJobExecutionActorState
+  case object CheckingJobStatus extends EngineJobExecutionActorState
+  case object PreparingJob extends EngineJobExecutionActorState
+  case object RunningJob extends EngineJobExecutionActorState
+
+  /** Commands */
+  sealed trait EngineJobExecutionActorCommand
+  case class Start(jobKey: BackendJobDescriptorKey) extends EngineJobExecutionActorCommand
+  case class Restart(jobKey: BackendJobDescriptorKey) extends EngineJobExecutionActorCommand
+
+  case class JobRunning(jobDescriptor: BackendJobDescriptor, backendJobExecutionActor: ActorRef)
+
+  def props(executionData: WorkflowExecutionActorData, factory: BackendLifecycleActorFactory,
+            initializationData: Option[BackendInitializationData]) = {
+    Props(new EngineJobExecutionActor(executionData, factory, initializationData))
+  }
+
+  /** Data */
+  case class EngineJobExecutionActorData(restarting: Boolean = false)
+}
+
+class EngineJobExecutionActor(executionData: WorkflowExecutionActorData,
```

Comment:
nm, I see you're just passing it through to the `JobPreparationActor`.  Not sure how I missed this before... 😦